### ISSUE TITLE
Add leader: false to Zero's /state ouput

### ISSUE THREAD

**Issue by fl-max** [dgraph-io/dgraph#5742]
## Experience Report

Note: Feature requests are judged based on user experience and modeled on [Go Experience Reports](https://github.com/golang/go/wiki/ExperienceReports). These reports should focus on the problems: they should not focus on and need not propose solutions.

### What you wanted to do
Have the `/state` output from Zero explicitly define a node as **non-leader** via `leader: false`. Currently, if a node is not a leader, the key is not present. Example:

**Current**:
```json
{
  "groups": {
    "1": {
      "members": {
        "1": {
          "id": "1",
          "groupId": 1,
          "addr": "alpha0:7080",
          **"leader": true,**
          "lastUpdate": "1592319336"
        },
        "2": {
          "id": "2",
          "groupId": 1,
          "addr": "alpha1:7080"
        },
        "3": {
          "id": "3",
          "groupId": 1,
          "addr": "alpha2:7080"
        }
      },
    }
  },
  "zeros": {
    "1": {
      "id": "1",
      "addr": "zero0:5080",
      **"leader": true**
    },
    "2": {
      "id": "2",
      "addr": "zero1:5080"
    },
    "3": {
      "id": "3",
      "addr": "zero2:5080"
    }
  },
}
```

**Proposal**:
```json
{
  "groups": {
    "1": {
      "members": {
        "1": {
          "id": "1",
          "groupId": 1,
          "addr": "alpha0:7080",
          "leader": true,
          "lastUpdate": "1592319336"
        },
        "2": {
          "id": "2",
          "groupId": 1,
          "addr": "alpha1:7080",
          **"leader": false**
        },
        "3": {
          "id": "3",
          "groupId": 1,
          "addr": "alpha2:7080",
          **"leader": false**
        }
      },
    }
  },
  "zeros": {
    "1": {
      "id": "1",
      "addr": "zero0:5080",
      "leader": true
    },
    "2": {
      "id": "2",
      "addr": "zero1:5080",
      **"leader": false**
    },
    "3": {
      "id": "3",
      "addr": "zero2:5080",
      **"leader": false**
    }
  },
}
```

### What you actually did
See above.

### Why that wasn't great, with examples

1. Doesn't explicitly tell the user that this node is **not** a leader; must be assumed.
2. `hasKey()` lookups are more process intensive
3. Dynamic schemas require more logic and are harder to maintain with upstream updates

**Comment by parasssh** [comment]
This is because the field is omitted if it is empty. 
` Leader  bool    `protobuf:"varint,4,opt,name=leader,proto3"` json:"leader,omitempty" `

**Comment by rahulgurnani** [comment]
The response json is constructed from json serialisation of protobuf object. As @parasssh mentioned if a bool is set to false it gets omitted during serialisation in proto3. From protobuf docs [here](https://developers.google.com/protocol-buffers/docs/proto3#default):

> Note that for scalar message fields, once a message is parsed there's no way of telling whether a field was explicitly set to the default value (for example whether a boolean was set to false) or just not set at all: you should bear this in mind when defining your message types. For example, don't have a boolean that switches on some behaviour when set to false if you don't want that behaviour to also happen by default. Also note that if a scalar message field is set to its default, the value will not be serialized on the wire.

**Comment by fl-max** [comment]
[including_default_value_fields](https://googleapis.dev/python/protobuf/latest/google/protobuf/json_format.html)

**Comment by NamanJain8** [comment]
Hi @fl-max, the issue has been addressed in #5966 . Thanks.
